<p>
  The <code>eval</code> function is a way to run arbitrary code at run-time. Generally it is considered to be very dangerous because it allows execution of arbitrary code. Its use thus is discouraged. If you have carefully verified that there is no other option than to use this construct, pay special attention not to pass any user-provided data into it without properly validating it beforehand.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
eval(code_to_be_dynamically_executed)
</pre>
