<p>
The following words may be used as keywords in future evolutions of the language, so using them as identifiers should be avoided to allow an easier adoption of those potential future evolutions:
<ul>
  <li>implements</li>
  <li>let</li>
  <li>private</li>
  <li>public</li>
  <li>yield</li>
  <li>interface</li>
  <li>package</li>
  <li>protected</li>
  <li>static</li>
</ul>

<p>Usage of these words as identifiers would produce an error in JavaScript strict mode code.</p>

<p>If ECMAScript6 parameter is set to true the following words will not raise any issue:</p>
</p>
<ul>
    <li>let</li>
    <li>yield</li>
</ul>

<h2>Noncompliant Code Example</h2>
<pre>
var package = document.getElementsByName("foo"); // Non-Compliant
</pre>


<h2>Compliant Solution</h2>
<pre>
var elements = document.getElementsByName("foo"); // Compliant
var someData = { package: true };                 // Compliant, as it is not used as an identifier here
</pre>
