<p>
  Giving both a variable and a function with the same name in a single scope should be avoided. This usage is confusing and should be avoided because it will be unclear whether the named variable refers to the function or to something else.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
var fun;
function fun() {
}
</pre>

<h2>Compliant Solution</h2>
<pre>
var fun = function fun() {
}
</pre>
